app/code/core/Mage/Core/functions.php

```diff
@@ -303,83 +296,49 @@ function is_dir_writeable($dir)
 
 function mahoGetComposerInstallationData(): array
 {
-    static $composerInstallationData = null;
-    if ($composerInstallationData !== null) {
-        return $composerInstallationData;
-    }
+    return \Maho\MahoAutoload::getInstalledModules(BP);
+}
```

Comment:
at this point I'd probably remove mahoGetComposerInstallationData function completely